### ISSUE TITLE
VSTest Coded UI Tests cannot download dependent files

### ISSUE THREAD

**Issue by acm74** [Microsoft/vsts-tasks#7246]
we have an existing MTM coded UI testing process that we are in the process of migrating to VSTS Build/Release.  

During test execution the test is designed to communicate with VSTS and download a relevant database backup file for use.  We have had this working, until the last couple of weeks.  Since then we consistently see "TF30063: You are not authorized to access https://********.visualstudio.com/." when using our older connector (which uses the SOAP API).  

Updating the connector to use the REST API (via the Microsoft.TeamFoundation.Client package) now produces the following error:

> Unit Test Adapter threw exception: 
> 
> Type is not resolved for member 'Microsoft.VisualStudio.Services.Common.VssUnauthorizedException,Microsoft.VisualStudio.Services.Common, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.

The VSTTAgent (qtagent32_40.exe) is from VS2015 (i.e. version 14).  So I think that explains the type not resolved error. 

Is it possible to upgrade the VSTTAgent to VS2017? If so, then how (we never explicitly installed Visual Studio on any of the test VMs, so I'm unsure at what point these files were installed)?

Why would we be seeing errors when attempting to communicate with VSTS?  We are using a PAT to authenticate to VSTS.  This works in a local environment, in our old MTM lab, and has previously worked in this environment. I have verified the PAT has not expired, and is scoped to all scopes.

**Comment by prawalagarwal** [comment]
Hi @acm74, Thanks for reporting. Please attach your run logs for us to understand your scenario better. You can reach out to devops_tools[at]microsoft[dot]com

**Comment by acm74** [comment]
@prawalagarwal the release logs have not been showing much of anything.  The error messages are appearing in stacktraces for test execution (either in .trx files, or sometimes in standard console output).

This is now becoming a more serious problem for us.  In other internal systems that we rely on, when we change the personal access token, we are seeing the same type of authorization exceptions (TF30063).

![testadaptorexception](https://user-images.githubusercontent.com/36503947/40354295-ca425d72-5db3-11e8-9775-c6ec3ee6c19c.PNG)
![testadaptorexception2](https://user-images.githubusercontent.com/36503947/40354296-ca636a76-5db3-11e8-90f2-eb101e88fde3.PNG)

**Comment by acm74** [comment]
@prawalagarwal we are seeing authorization errors in more of our system - this has also been reported separately at: https://developercommunity.visualstudio.com/content/problem/258204/login-using-new-pat-fails.html

**Comment by prawalagarwal** [comment]
Hi @acm74 Firstly, please dont take dependency on Microsoft.TeamFoundation.Client packaged with Visual studio. There is a nuget package specifically made to communicate to VSTS [link](https://www.nuget.org/packages/Microsoft.TeamFoundationServer.ExtendedClient/). Please use this nuget package. 

If this doesnt solve your issue please attach a sample source code which is failing to connect to VSTS. You can reach out at devops_tools[at]microsoft[dot]com

**Comment by acm74** [comment]
@prawalagarwal I have tried to use both the Microsoft.TeamFoundation.Client NuGet package and the ExtendedClient package.  Same problem with either package.

**Comment by prawalagarwal** [comment]
Please attach your logs with system.debug set as true in release variables and a sample code which is failing to communicate. You can send them to devops_tools[at]microsoft[dot]com

**Comment by acm74** [comment]
@prawalagarwal release logs are 32Mb - this is too big to upload to GitHub.  Where can I send it?

**Comment by prawalagarwal** [comment]
@acm74 We did not get the mail from you. It might have been filtered as spam due to the attachment size.  Is it possible to share both logs and the sample code on a share and provide us the link on devops_tools@microsoft.com

**Comment by acm74** [comment]
@prawalagarwal I will try again later today.  Please note that the related issue with authorization has been fixed, and I was able to get a test run started.  I have had to roll back our code changes to remove the use of the VSTS REST APIs (via the Microsoft.TeamFoundation.Client NuGet package).

**Comment by Nekobravo** [comment]
![image](https://user-images.githubusercontent.com/23462665/40460574-718b6caa-5ed5-11e8-8262-a942ca10c6bc.png)
![image](https://user-images.githubusercontent.com/23462665/40460604-91717e06-5ed5-11e8-9e2e-d562b13cf69f.png)
Help!!!

**Comment by prawalagarwal** [comment]
@Nekobravo As I mentioned on the other thread, please create a separate issue or reach out to us at devops_tools@microsoft.com

**Comment by prawalagarwal** [comment]
@acm74 Please let us know if you are unblocked. Also it will help if you could mention the steps which helped you unblock.

**Comment by acm74** [comment]
@prawalagarwal we are currently still experiencing issues with unavailability of VSTS.  We are seeing HTTP requests timing out, service unavailable errors, and service unavailable for anonymous authentication errors.

**Comment by acm74** [comment]
@prawalagarwal we have a work-around for the issue with the test adaptor throwing an exception.  We can get tests running now (when VSTS authentication and authorization is working).  

In the longer term we would like to start using the VSTS REST APIs, so any insight or help you can provide would be appreciated.  I am still trying to find out just how the dependencies to VS2015 were installed on our test VMs.

**Comment by acm74** [comment]
@prawalagarwal I will see what I can do.  Our codebase is not so easy to work with, and unfortunately I don't have the time to try and put together a proof of concept.  

In the meantime, I am still seeing issues with VSTS itself.  All my attempts today to configure new interactive test agents are failing with "TF400813: resource not available for anonymous access".

**Comment by acesiddhu** [comment]
@acm74 ensure that the PAT token you are using is still valid. you will generally see this error when PAT token has expired

**Comment by acm74** [comment]
@acesiddhu sorry about the delay in replying - I've had to work on other issues in the past couple of days.  The PAT is still valid. I updated the agent version being used (from 2.129.0 to 2.133.3), and this seems to have fixed the problem.

**Comment by prawalagarwal** [comment]
Thanks for updating us. Closing this issue.
